### PR TITLE
updated button color

### PR DESCRIPTION
**Pull request by james-muriithi**
most of the changes in the theme.scss were made by prettier, i added from line 453 and white variable on line 102

### CODE REVIEW

File: frontend/src/components/ChallengeCard.vue [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+          </div>
+          <div class="col-6 text-end">
+            <small class="text-muted">{{
+              challenge.programmingLanguage
+            }}</small>
+          </div>
+        </div>
       </b-col>
-      <b-col cols="3">
+      <!-- <b-col cols="3">
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:32]: Don't commit commented-out code!

File: frontend/src/components/ChallengeCard.vue [hunk 2]
```diff
@@ -32,15 +47,45 @@ export default {
       type: Object,
     },
   },
-  methods: {
-    redirectToSingleChallenge(id) {
-      this.$router.push({ name: "clg-single", params: { id } });
+  computed: {
+    challengeTitle(){
+      if (this.challenge.title.length > 20) {
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:52]: A ternary might be cleaner. What do you think?

File: frontend/src/components/ChallengeCard.vue [hunk 3]
```diff
@@ -32,15 +47,45 @@ export default {
...
+
+      return this.challenge.title;
     },
-  },
-  mounted() {
-    if (this.challenge.title.length > 23) {
-      this.challenge.title = this.challenge.title.slice(0, 20) + "...  ";
+    challengeDescription() {
+      if (this.challenge.text.length > 500) {
+        return this.challenge.text.slice(0, 500) + `... <b-link to="${this.challengeLink}" class="more text-primary">More</b-link>`;
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:60]: the link Seems malformed. Have you checked that it works?

File: frontend/src/components/ChallengeCard.vue [hunk 4]
```diff
@@ -1,25 +1,40 @@
...
     "
     cols="12"
     @click="redirectToSingleChallenge(challenge.id)"
   >
     <b-row class="pt-1">
       <b-col>
-        <h5>{{ challenge.title }}</h5>
+        <h5 class="challenge-title">{{ challengeTitle }}</h5>
+        <p class="text-muted">26 min ago</p>
+        <p class="challenge-description" v-html="challengeDescription"></p>
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:20]: Using v-html here Seems hacky. Just put in the b-link in and add a v.if

File: frontend/src/components/ChallengeCard.vue [hunk 5]
```diff
@@ -32,15 +47,45 @@ export default {
...
-    if (this.challenge.title.length > 23) {
-      this.challenge.title = this.challenge.title.slice(0, 20) + "...  ";
+    challengeDescription() {
+      if (this.challenge.text.length > 500) {
+        return this.challenge.text.slice(0, 500) + `... <b-link to="${this.challengeLink}" class="more text-primary">More</b-link>`;
+      }
+
+      return this.challenge.text;
+    },
+    challengeLink(){
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:65]: Good!

File: frontend/src/components/ChallengeCard.vue [hunk 6]
```diff
@@ -32,15 +47,45 @@ export default {
...
+  methods: {
+    redirectToSingleChallenge() {
+      this.$router.push(this.challengeLink);
+    },
+  },
 };
 </script>
+
+<style lang="scss" scoped>
+.challenge {
```
Comment (Wittmaxi) [inline frontend/src/components/ChallengeCard.vue:78]: I'm not a huge fan of having a lot of CSS - You decide what you want, but it's important that the designs stay consistent within Talentrator and Prograchall

**Review by Wittmaxi — CHANGES_REQUESTED** [review]
This is what code reviews are for! You have done excellent work, now we need to align our visions of good code and good design so we can work as one Team 😁

File: frontend/src/assets/scss/partials/_animations.scss [hunk 7]
```diff
@@ -0,0 +1,66 @@
+// shake error keyframe
+@keyframes shakeError {
```
[the remark this answers is not in the document]
Comment (james-muriithi) [reply frontend/src/assets/scss/partials/_animations.scss:2]: They were in the theme.scss file, i just put them in a separate file

File: frontend/src/components/ChallengeCard.vue (hunk 2 above)
Comment (james-muriithi) [reply frontend/src/components/ChallengeCard.vue:52]: yes

File: frontend/src/assets/scss/partials/_animations.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (james-muriithi) [reply frontend/src/assets/scss/partials/_animations.scss:2]: removed

File: frontend/src/components/ChallengeCard.vue (hunk 3 above)
Comment (james-muriithi) [reply frontend/src/components/ChallengeCard.vue:60]: yes it was malformed, thanks

File: frontend/src/components/ChallengeCard.vue (hunk 6 above)
Comment (james-muriithi) [reply frontend/src/components/ChallengeCard.vue:78]: okay

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 8]
```diff
@@ -73,22 +73,32 @@
...
-            class="form__input"
-            :class="{
-              'mb-2': !$v.form.text.$error,
-            }"
-          />
-          <div v-if="$v.form.text.$dirty">
-            <div class="error mb-2" v-if="!$v.form.text.required">
-              Please input your answer
+          <b-button variant="secondary" size="sm" class="px-1 py-1 mb-1 text-capitalize" @click="previewMarkdown = !previewMarkdown"
+            >{{previewMarkdown ? "Continue Editing": "Preview"}}
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:77]: these brackets are misformatted

**Review by Wittmaxi — APPROVED** [review]
Very good! :)

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 9]
```diff
@@ -139,6 +193,38 @@ export default {
       this.form = { email: "", nickname: "", text: "" };
       this.$v.$reset();
       this.submitting = false;
+      this.previewMarkdown = false;
+    },
+    categorizeCodeSnippetsAndText(text) {
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:198]: Code is reused from Comment.vue

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 10]
```diff
@@ -111,17 +141,41 @@ import {
...
+  components: {
+    CodeEditor,
+  },
   data: () => ({
     form: {
       email: "",
       nickname: "",
       text: "",
     },
+    langToEditor: {
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:157]: Code is reused from Comment.vue

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 11]
```diff
@@ -73,20 +73,50 @@
...
+            <div v-if="$v.form.text.$dirty">
+              <div class="error mb-2" v-if="!$v.form.text.required">
+                Please input your answer
+              </div>
+            </div>
+          </div>
+
+          <div v-else>
+            <div
+              v-for="(description, idx) in categorizeCodeSnippetsAndText(
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:106]: Code is reused from Comment.vue

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 12]
```diff
@@ -73,20 +73,50 @@
...
+          </b-button>
+
+          <div v-if="!previewMarkdown">
+            <b-form-textarea
+              rows="3"
+              v-model="form.text"
+              name="text"
+              placeholder="Leave your answer here..."
+              v-model.trim="$v.form.text.$model"
+              class="form__input"
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:92]: does `form__input` exist?

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue [hunk 13]
```diff
@@ -111,17 +141,41 @@ import {
...
+      swift4: "swift",
+      go: "golang",
+      rust: "rust",
+      scala: "scala",
+    },
     submitting: false,
+    previewMarkdown: false,
   }),
   methods: {
+    markdownToHtml(markdown) {
```
Comment (Wittmaxi) [inline frontend/src/pages/challenges/single-challenge/FormulateComment.vue:176]: This same challenge was solved differently in Comment.vue

**Review by Wittmaxi — CHANGES_REQUESTED** [review]
Hey James!
Great work, Stuff seems to be working great! :)
Clean code is super important! This is why it's important that every piece of code follows the SRP from the beginning! How about you work with Rishabh on coding one Component that displays Markdown with our code-editor?

File: frontend/src/pages/challenges/single-challenge/FormulateComment.vue (hunk 12 above)
Comment (james-muriithi) [reply frontend/src/pages/challenges/single-challenge/FormulateComment.vue:92]: there is one style for transition

**Review by Wittmaxi — APPROVED** [review]

**Pull request merged**